internal/verifier/verifier.go

```diff
@@ -78,10 +78,16 @@ func (verifier *ReproducibleProvenanceVerifier) Verify(provenanceFilePath string
 		return fmt.Errorf("couldn't build the binary: %v", err)
 	}
 
-	// The provenance is valid, therefore `expectedBinaryHash` is guaranteed to be non-empty.
-	expectedBinaryDigest := provenance.GetBinarySHA256Digest()
+	var actual ProvenanceIR
+	actual.FromAmber(provenance)
 
-	if err := buildConfig.VerifyBinarySHA256Digest(expectedBinaryDigest); err != nil {
+	expectedBinaryDigest, err := buildConfig.ComputeBinarySHA256Digest()
```

Comment:
I don't quite see the benefit of this expansion. Can this logic be refactored to a `FromBuildConfig` function?